canary/webrtc-c/src/Peer.cpp

```diff
@@ -597,17 +598,18 @@ STATUS Peer::writeFrame(PFrame pFrame, MEDIA_STREAM_TRACK_KIND kind)
     }
     for (auto& transceiver : transceivers) {
         retStatus = ::writeFrame(transceiver, pFrame);
-        if (retStatus == STATUS_SRTP_NOT_READY_YET) {
-            // do nothing
-            retStatus = STATUS_SUCCESS;
+        CHK (retStatus == STATUS_SRTP_NOT_READY_YET || retStatus == STATUS_SUCCESS, retStatus);
+
+        if (this->firstFrame && this->isMaster) {
```

Comment:
Oh, we might need to think about this in case of retStatus == STATUS_SRTP_NOT_READY_YET. Perhaps we need to do the differentiation here and reset the value latest before exiting.

if (this->firstFrame && this->isMaster) {
...
}
retStatus = STATUS_SUCCESS; // Reset the status

Reply:
Oops! My bad. Completely missed the case. I fixed it now.